### DISCUSSION TITLE
FerretDB vs Mongodb comparison

### DISCUSSION THREAD

**Discussion by snail2sky** [FerretDB/FerretDB#4400] [General]
## High availability
- Mongodb supports replicas, sharding, and sharded replica sets. How does FerretDB do this?
- How does FerretDB perform failover after a failure occurs? Can it be switched quickly like Mongodb?
- How to make a backup and how to perform backup and recovery? Is it production ready?

## performance
-  Are there any stress tests and how does FerretDB compare to mongodb?

**Comment by AlekSi** [comment]
Hi, sorry for a long delay.

FerretDB does not yet support replication and sharding. It is unclear if this should be implemented in the FerretDB itself or delegated to the PostgreSQL. See #178 for the tracking issue.

The answer above applies to the failover as well. FerretDB is stateless, so you might use PostgreSQL replication and failover solutions.

Backup and restore could be performed using standard MongoDB tools: mongodump/mongorestore and mongoexport/mongoimport. See https://docs.ferretdb.io/migration/migrating-from-mongodb/.

We don’t provide public performance testing results yet. The next major of FerretDB (not public yet) is quite close to MongoDB in that area, but it depends on your usage patterns.

**Reply by icereed** [reply]
@AlekSi is the v2 already in development in a hidden breach or repo? Asking as since you mentioned it’s not available to the public.

**Reply by AlekSi** [reply]
Yes, v2 is currently in development in a private repo. It is built on top of the PostgreSQL extension and is in the alpha stage. We plan to release it under the Apache 2.0 license in several months.

**Reply by icereed** [reply]
That’s fantastic news!

**Comment by snail2sky** [comment]
Okay, thank you for your answer, I will continue to follow the official information

**Comment by irelance** [comment]
I make some test and result here:
https://github.com/irelance/document-database-benchmark-result/tree/master/2025-03-04
